src/Microsoft.Data.SqlClient/tests/FunctionalTests/SqlConnectionBasicTests.cs

```diff
@@ -23,6 +25,35 @@ public void ConnectionTest()
             connection.Open();
         }
 
+        [Fact]
+        public void ConnectionTestWithCultureTH()
+        {
+            // Save current cultures
+            CultureInfo savedCulture = Thread.CurrentThread.CurrentCulture;
+            CultureInfo savedUICulture = Thread.CurrentThread.CurrentUICulture;
+
+            try
+            {
+                Thread.CurrentThread.CurrentCulture = new CultureInfo("th-TH");
+                Thread.CurrentThread.CurrentUICulture = new CultureInfo("th-TH");
+
+                using TestTdsServer server = TestTdsServer.StartTestServer();
+                using SqlConnection connection = new SqlConnection(server.ConnectionString);
+                connection.Open();
+                Assert.Equal(ConnectionState.Open, connection.State);
+            }
+            catch (Exception e)
+            {
+                Assert.False(true, e.Message);
+            }
```

Comment:
Just throw the exception, it's not needed to be assessed.
```suggestion
            catch
            {
                throw;
            }
```